[hzk-lab] iP

hzk-lab opened the pull request:
# Donk
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

## 🚀 Your Personal AI Task Manager!
A powerful, lightweight, and smart task manager designed to simplify your workflow and boost productivity.😉

## 📌 Features at a Glance

1. 📝 Task Management: Add, remove, and list tasks effortlessly.
2. ⏳ Deadlines & Events: Never miss important dates again!
3.  🔍 Smart Search: Find tasks by keywords or dates.
4.  💾 Persistent Storage: Saves tasks automatically to disk.
5.  🎨 JavaFX GUI: Beautiful and interactive interface.

AND it is **FREE**!

## 🔥 How to Get Started
1️⃣ Download the latest version from [Releases](https://github.com/your-repo/releases)
2️⃣ Run the app using:
```sh 
java -jar donk.jar
```

3️⃣ Add your first task

```sh
todo Read JavaFX tutorial
```
4️⃣ Enjoy a seamless task management experience!

## 🛠️ Upcoming Enhancements

 - Recurring tasks
 - Google Calendar sync
 - JavaFX GUI added!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hzk-lab committed:
Level-0

hzk-lab committed:
Add support for echo

hzk-lab committed:
Add support for list

hzk-lab committed:
Add support for marking

hzk-lab committed:
Add support for Todos, Deadlines and Events

hzk-lab committed:
Add Close to Scanner to avoid wastage

hzk-lab committed:
A-TextUiTesting

hzk-lab committed:
Add DonkException and try-catch blocks

hzk-lab committed:
Add testing for DonkException, and modify Donk accordingly

hzk-lab committed:
Add delete support and corresponding Exceptions

hzk-lab committed:
Add delete support and corresponding Exceptions

hzk-lab committed:
Add saving function

hzk-lab committed:
Merge Level-7

hzk-lab committed:
Fix file storage format

hzk-lab committed:
Fix file storage format

hzk-lab committed:
More Level-7 improvements

hzk-lab committed:
Fix storage format

hzk-lab committed:
Save changes before switching to branch-Level-8

hzk-lab committed:
Resolve merge conflict

hzk-lab committed:
Save changes before switching to branch-Level-8

hzk-lab committed:
Save changes before switching to branch-Level-8

hzk-lab committed:
Merge branch 'master' into branch-Level-8

The development shuold be on branch-level-8.

hzk-lab committed:
Add date search for deadlines/events

hzk-lab committed:
Merge Level-8 into master

hzk-lab committed:
Add OOP programming

hzk-lab committed:
Add package

hzk-lab committed:
Save progress before switching to add-gradle-support

hzk-lab committed:
Merge branch 'add-gradle-support'
Build Gradle

deborahluyx commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,32 @@
+    public void showWelcome() {
+        System.out.println("____________________________________________________________\n"
+                + "Hello! I'm Donk\n"
+                + "What can I do for you?\n");
```
I like how you broke up this print statement into multiple lines to prevent it from being too long!

hzk-lab replied:
Thanks!

deborahluyx commented on src/main/java/Storage.java:
```diff
@@ -1,4 +1,5 @@
 import java.io.*;
```
Should the java.io classes be listed explicitly instead?

hzk-lab replied:
Good suggestion! I have changed the import format!

deborahluyx commented on src/main/java/donk/Parser.java:
```diff
@@ -0,0 +1,79 @@
+            case "event":
+                String[] eventParts = words[1].split(" /from | /to ");
+                if (eventParts.length < 3) {
+                    throw new DonkException("Invalid event format! Use: event <task> /from <start> /to <end>");
```
I think there is no indentation for `case` clauses. Perhaps you can try configuring your IDE to follow this style instead?

deborahluyx commented on src/main/java/donk/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+import java.time.format.DateTimeParseException;
+
+public class TaskList {
+    private final List<Task> tasks;
```
Should this variable be in screaming snake case instead since it's a constant?

hzk-lab replied:
I don’t think so. Since the variable tasks is not a compile-time constant; rather, it is an immutable reference to a list that can still be modified (I.e. new elements can be added or removed).

Only variables with “final static” should be in screaming snake case, which is not in this case.

deborahluyx commented on src/main/java/donk/Storage.java:
```diff
@@ -0,0 +1,101 @@
+        this.filePath = file_path;
+    }
+
+    // Load tasks from file (handling corruption)
```
Do remember that JavaDoc is a mandatory step in Week 3's iP tasks. Perhaps you can switch to JavaDocs soon, providing more information about the function such as its parameters and what it returns?

hzk-lab replied:
Yes, I have switched to JavaDoc by now!

deborahluyx reviewed (commented):
Minor adjustments needed to adhere to coding standard, but besides that looks pretty good overall. Keep up the good work! 👍🏻

hzk-lab committed:
Add JUnit tests

hzk-lab committed:
Add Jar file

hzk-lab committed:
Add header comments

hzk-lab committed:
Add features for task name search

hzk-lab committed:
Resolved merge conflict in TaskList.java

bryantjandra commented on src/main/java/seedu/donk/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     *
+     * @param file_path The file path where tasks are stored.
+     */
+    public Storage(String file_path) {
```
The parameter name file_path uses an underscore. Shouldn't it perhaps be camel-cased instead?

bryantjandra commented on src/main/java/seedu/donk/parseDate.java:
```diff
@@ -0,0 +1,49 @@
+ * It provides methods for converting date strings into a standard format
+ * and validating the chronological order of start and end dates.
+ */
+public class parseDate {
```
I think the class name `parseDate` should be changed to `ParseDate` as it doesn’t follow the Pascal-case convention for class names.

bryantjandra commented on src/main/java/seedu/donk/TaskList.java:
```diff
@@ -0,0 +1,211 @@
+ */
+public class TaskList {
+
+    private final List<Task> tasks;
```
I like how you distinguished clearly between single-valued and multi-valued variables using `tasks` instead of `task` here.

broyantt reviewed (commented):
LGTM from a coding-quality perspective. Just a few nits as mentioned! Overall, I found your code easy to read and understand.

karthu0301 commented on src/main/java/seedu/donk/Storage.java:
```diff
@@ -0,0 +1,131 @@
+import seedu.donk.task.Task;
+import seedu.donk.task.ToDo;
+
+import java.io.*;
```
Try not to use wildcard imports, but otherwise method descriptions are all pretty clear!

karthu0301 commented on src/main/java/seedu/donk/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     * @param line The task entry in the file.
+     * @return The corresponding {@code Task} object, or {@code null} if the line is corrupted.
+     */
+    private Task parseTask(String line) {
```
Maybe consider moving this method to your parser class?

karthu0301 commented on src/main/java/seedu/donk/command/Command.java:
```diff
@@ -0,0 +1,15 @@
+package seedu.donk.command;
```
It's nice that you separated all your command classes, makes your code much more readable and easier to debug!

karthu0301 commented on src/test/java/seedu/donk/StorageTest.java:
```diff
@@ -0,0 +1,73 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+class StorageTest {
+    private static final String TEST_FILE_PATH = "./test_data/donk_test.txt";
```
Naming for constants is correct in screaming snake case.

hzk-lab committed:
Add GUI features

hzk-lab committed:
Fix minor bugs

hzk-lab committed:
Fix minor bugs

hzk-lab committed:
Change message passing

hzk-lab committed:
Fit git standard

hzk-lab committed:
Improve code quality

hzk-lab committed:
Add assertions

hzk-lab committed:
Merge branch-A-Assertions

hzk-lab committed:
Resolve merge conflict between master and branch-A-CodeQuality

hzk-lab committed:
Implement sorting features

hzk-lab committed:
Add first ReadMe file

hzk-lab committed:
Fix code quality

hzk-lab committed:
Merge pull request #2 from hzk-lab/branch-A-Assertions

Add first ReadMe file

hzk-lab committed:
Merge pull request #3 from hzk-lab/branch-A-CodeQuality

Fix code quality

hzk-lab committed:
Fix jpeg format

hzk-lab committed:
Update documentation README

hzk-lab committed:
Fix storage bugs

hzk-lab committed:
Fix storage bugs

hzk-lab committed:
Fix storage bugs

hzk-lab committed:
Add Ui image

hzk-lab committed:
Add README file

hzk-lab committed:
fix: Resolve issue where images were not displaying in GUI

- Replaced WebP images with PNG due to JavaFX WebP incompatibility.
- Moved images to `src/main/resources/seedu/donk/images/` to match classpath.
- Used `this.getClass().getResourceAsStream("/seedu/donk/images/Donk.png")`
  to load images correctly in packaged JAR.
- Verified working directory paths to ensure correct loading.

hzk-lab committed:
Complete UI settings

hzk-lab committed:
Fix minor bugs

hzk-lab committed:
Fix minor bugs

hzk-lab committed:
Fix event time bugs
